cmd/server-main.go

```diff
@@ -1037,6 +1040,17 @@ func serverMain(ctx *cli.Context) {
 		}
 		fmt.Println("======")
 	}
+	go func() {
+		<-startDoneCh
+		// Object layer is initialized then print StorageInfo.
+		objAPI := newObjectLayerFn()
+		if objAPI != nil {
+			printStorageInfo(objAPI.StorageInfo(GlobalContext, true))
+		}
+		for _, warn := range warnings {
+			logger.Info(warn)
+		}
+	}()
```

Comment:
You could have moved this code inside the above goroutine itself - there is no reason to have <-startupDoneCh, it won't change anything as such. 

Move DEBUG to print at the very end. 

```suggestion
	go func() {
		<-startDoneCh
		// Object layer is initialized then print StorageInfo.
		objAPI := newObjectLayerFn()
		if objAPI != nil {
			printStorageInfo(objAPI.StorageInfo(GlobalContext, true))
		}
		for _, warn := range warnings {
			logger.Info(warn)
		}
	}()
```

Reply:
the change is to print all warnings at the end after all startup banner stuff is done - since warnings are logged in several goroutines, msg ordering is not guaranteed

Reply:
No this goroutine and channel is unnecessary.

Reply:
is the warning [here](https://github.com/minio/minio/blob/master/cmd/server-main.go#L736) not going to print before the go routine that prints the startup banner [here](https://github.com/minio/minio/blob/master/cmd/server-main.go#L993). The ask was to avoid interspersing warnings in b/w startup messages

Reply:
Correct you can do that without the channel at the same place where  you have the `startupCh <- struct{}{}` 

instead of this just write the entire code that you are doing in goroutine.

Reply:
Done